app/src/containers/b2b/AccountMain.tsx

```diff
@@ -212,13 +220,15 @@ export default class AccountMain extends React.Component<RouteComponentProps<Acc
   subAccountData(data) {
     this.setState({
       accountName: data.name,
-      externalId: data['external-id'],
+      externalId: data._accountmetadata ? data._accountmetadata[0]['external-id'] : null,
       registrationNumber: data['registration-id'],
       legalName: data['legal-name'],
       associates: data._associateroleassignments[0]._element ? data._associateroleassignments[0]._element.map(element => ({ associate: element._associate[0], roles: element._roleinfo[0] })) : [],
       addAssociateUri: data._associateroleassignments[0]._associateform[0]._addassociateaction[0].self.uri,
       addSubAccountUri: data._subaccounts[0]._accountform[0].self.uri,
+      addSubAccountSellerAdmin: typeof data._subaccounts[0]._accountform[0]['external-id'] !== 'undefined',
```

Comment:
Did you intend to get the type of this object here, or did you want the actual object?

Reply:
Did you test if no subaccounts are returned? Does this fail?

Reply:
same as above.

Reply:
Discussed in person. The intention makes sense.